tests/UnusualContentTest.php

```diff
@@ -104,4 +104,34 @@ public function testAddingNewlineAtEOF() {
     $this->assertContains('+bar', $hunk);
     $this->assertContains('\ No newline at end of file', $hunk);
   }
+
```

Comment:
@fredemmott added tests here instead of in their own file. Seemed like it was appropriate.